components/insight/SRC/org/openmicroscopy/shoola/env/data/views/calls/ThumbnailLoader.java

```diff
@@ -62,395 +59,305 @@
+                        }
+                    }
+                };
+
+                add(call);
+            }
+        }
+    }
+
+    private IConfigPrx getConfigService() throws DSOutOfServiceException {
+        if (configService == null) {
+            configService = context.getGateway()
+                    .getConfigService(ctx);
+        }
+        return configService;
+    }
+
+    private void handleBatchCall(ThumbnailStorePrx store, PixelsData pxd, long userId) {
+        // If image has pyramids, check to see if image is ready for loading as a thumbnail.
+        try {
+            Image thumbnail;
+            byte[] thumbnailData = loadThumbnail(store, pxd, userId);
+            if (thumbnailData == null || thumbnailData.length == 0) {
+                // Find out why the thumbnail is not ready on the server
+                if (requiresPixelsPyramid(pxd)) {
+                    thumbnail = determineThumbnailState(pxd);
+                } else {
+                    thumbnail = Factory.createDefaultThumbnail("Loading");
+                }
+            } else {
+                thumbnail = WriterImage.bytesToImage(thumbnailData);
+            }
+            // Convert thumbnail to whatever
+            currentThumbnail = new ThumbnailData(pxd.getImage().getId(),
+                    thumbnail, userId, true);
+        } catch (Exception e) {
+            context.getLogger().error(this, e.getMessage());
+        }
+    }
+
+    private PixelsData dataObjectToPixelsData(DataObject image) {
+        return image instanceof ImageData ?
+                ((ImageData) image).getDefaultPixels() :
+                (PixelsData) image;
+    }
+
+    private Image determineThumbnailState(PixelsData pxd)
+            throws DSOutOfServiceException, ServerError {
+        RawPixelsStorePrx rawPixelStore = context.getGateway()
+                .getPixelsStore(ctx);
+        try {
+            // This method will throw if there is an issue with the pyramid
+            // generation (i.e. it's not finished, corrupt)
+            rawPixelStore.setPixelsId(pxd.getId(), false);
+        } catch (omero.MissingPyramidException e) {
+            // Thrown if pyramid file is missing, then we know the thumbnail still has
+            // to be generated in a short time
+            return Factory.createDefaultThumbnail("Loading");
+        } catch (ResourceError e) {
+            context.getLogger().error(this, new LogMessage("Error getting pyramid from server," +
+                    " it might be corrupt", e));
+        }
+        return Factory.createDefaultThumbnail("Error");
+    }
+
+    private ThumbnailStorePrx getThumbnailStore(PixelsData pxd) throws DSAccessException,
+            DSOutOfServiceException, ServerError {
+        // System.out.println(image.getId());
+        ThumbnailStorePrx store = service.createThumbnailStore(ctx);
+        if (!store.setPixelsId(pxd.getId())) {
+            store.resetDefaults();
```

Comment:
This line is no longer covered in a try/catch block as it was previously. (cF. https://github.com/openmicroscopy/openmicroscopy/pull/5827/files?utf8=%E2%9C%93&diff=unified&w=1#diff-cc769d8303d8c7e9b3f1a015e3d7abfdL133)

This leads to:

```
   serverExceptionClass = "ome.conditions.ApiUsageException"
    message = "Unable to reset rendering settings in a read-only group for Pixels set id:149552"
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at IceInternal.BasicStream.createUserException(BasicStream.java:2779)
	at IceInternal.BasicStream.access$300(BasicStream.java:14)
	at IceInternal.BasicStream$EncapsDecoder10.throwException(BasicStream.java:3298)
	at IceInternal.BasicStream.throwException(BasicStream.java:2291)
	at IceInternal.OutgoingAsync.throwUserException(OutgoingAsync.java:399)
	at omero.api.ThumbnailStorePrxHelper.end_resetDefaults(ThumbnailStorePrxHelper.java:3614)
	at omero.api.ThumbnailStorePrxHelper.resetDefaults(ThumbnailStorePrxHelper.java:3501)
	at omero.api.ThumbnailStorePrxHelper.resetDefaults(ThumbnailStorePrxHelper.java:3488)
	at org.openmicroscopy.shoola.env.data.views.calls.ThumbnailLoader.getThumbnailStore(ThumbnailLoader.java:301)
	at org.openmicroscopy.shoola.env.data.views.calls.ThumbnailLoader.access$000(ThumbnailLoader.java:70)
	at org.openmicroscopy.shoola.env.data.views.calls.ThumbnailLoader$1.doCall(ThumbnailLoader.java:223)
```